src/main/java/com/stripe/model/EventData.java

```diff
@@ -10,6 +15,137 @@
 @Setter
 @EqualsAndHashCode(callSuper = false)
 public class EventData extends StripeObject {
+  // Original properties
   StripeObject object;
   Map<String, Object> previousAttributes;
+
+  // Augmented properties during deserialization
+  @Setter(AccessLevel.NONE)
+  JsonElement rawJson;
+  @Getter(AccessLevel.NONE)
+  @Setter(AccessLevel.NONE)
+  String apiVersion;
+
+  /**
+   * Stripe object {@link EventData#object} is intentionally left out during
+   * initialization. Its deserialization from {@link EventData#rawJson} is deferred to
+   * user instead. See {@link EventData#safeDeserialize()} and
+   * {@link EventData#forceDeserialize()}.
+   * @param rawJson             raw JSON for {@code object} field that can be deserialized into
+   *     StripeObject.
+   * @param previousAttributes  previous attributes of event data (has no further
+   *     deserialization concerns)
+   * @param apiVersion          api version of the event data, used to determine whether the
+   *     deserialization to {@code object} field is safe.
+   */
+  EventData(JsonElement rawJson,
+            Map<String, Object> previousAttributes,
+            String apiVersion) {
+    this.previousAttributes = previousAttributes;
+    this.rawJson = rawJson;
+    this.apiVersion = apiVersion;
+  }
+
+  /**
+   * Safe deserialize raw JSON into StripeObject. This operation mutates the state, and the
+   * successful result can be accessed via {@link EventData#getObject()}. Matching API
+   * version between the current integration and the event is necessary to guarantee safe
+   * deserialization.
+   * @return whether deserialization has been successful.
+   */
+  public boolean safeDeserialize() {
+    if (!apiVersionMatch()) {
```

Comment:
it will also return nothing when there's api version mismatch between the "pinned" version and the event version